### PR TITLE
[one-cmds] Add one-build_004 test

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit adds one-build_004 test. This test runs `one-import-tf` and
`one-codegen` at once with `one-build`.

Related: #5375 
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

**Comment by mhs4670go** [comment]
@seanshpark In our offline talk, we've decided to install `dummy-codegen` to `bin` folder and exclude it when packaging. But actually on second thoguht, when we just untar `nncc-1.12.0.tar.gz
` and `nncc-test-x.x.x.tar.gz` rather than building it, it would be a problem as well because there would be no `dummy-codegen` in `bin` file.

Speaking of which, how about revising this `one-cmds` test. The list of points to revise are like this.
- Integrate `one-cmds` test to Cmake ctest : it'd solve above issue and we'd can run `one-cmds` test in CI as well.
- Remove dependency between tests : Some test(e.g. `one-quantize_001.test`) doesn't create test material from scratch, which might occur problem sometimes.
- Create working directory : this is a bit similar with the second one. [create `working_dir` -> pushd `working_dir` -> test -> popd `working_dir`] process will make the tests more clear(?) and repeatable. Test output will be just a log file.

It is just my suggestion:) Please feel free to give me your opinion.

**Comment by seanshpark** [comment]
> it would be a problem as well because there would be no dummy-codegen in bin file.

I was thinking to put `dummy-codegen` in `nncc-test-x.x.x.tar.gz`

> Integrate one-cmds test to Cmake ctest

Purpose of this test is with release package. I'm not sure about this with Cmake test

**Comment by mhs4670go** [comment]
@seanshpark 
> Purpose of this test is with release package

got it.

> I was thinking to put `dummy-codegen` in `nncc-test-x.x.x.tar.gz`

Then, I will install `dummy-codegen` to `test` folder. If so, there is no need to exclude it from packaging like `tflchef` does. And, I need to copy `dummy-codegen` to `bin` folder before `one-codegen` test and remove it at the end of the test.

**Comment by mhs4670go** [comment]
@seanshpark PTAL:)

File: compiler/one-cmds/dummy-driver/src/dummy-compile.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  // ./dummy-compile -o ${OUTPUT_NAME} ${INPUT_NAME}
+  if (argc != 4)
+    return EXIT_FAILURE;
+
+  std::string opt_o{"-o"};
+  std::string output_name{argv[1]};
+  if (opt_o != output_name)
+    return EXIT_FAILURE;
+
+  std::string input_name{argv[2]};
```
Comment (seanshpark) [inline compiler/one-cmds/dummy-driver/src/dummy-compile.cpp:35]: please fix this variable name to `output_name`

File: compiler/one-cmds/dummy-driver/src/dummy-compile.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * dummy-compile only tests its interface rather than its functionality.
+ *
+ * ./dummy-compile -o ${OUTPUT_NAME} ${INPUT_NAME}
+ *
+ * NOTE argv[3](INPUT_NAME) not used as it is a dummy param.
```
Comment (seanshpark) [inline compiler/one-cmds/dummy-driver/src/dummy-compile.cpp:22]: Maybe `not used as this is a dummy program.`

Comment (mhs4670go) [reply compiler/one-cmds/dummy-driver/src/dummy-compile.cpp:22]: @seanshpark Actually, it's kinda funny,, it(argv[3]) is a dummy param("argument" might be more accurate) of dummy program..

Comment (seanshpark) [reply compiler/one-cmds/dummy-driver/src/dummy-compile.cpp:22]: Hm... maybe `NOTE argv[3](INPUT_NAME) is not used here`

File: compiler/one-cmds/tests/one-build_004.test [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+trap trap_err_onexit ERR
+
+configfile="one-build_004.cfg"
+outputfile="sample.tvn"
+
+rm -rf ${outputfile}
+
+# copy dummy-compile to bin folder
+cp dummy-compile ../bin/dummy-compile
```
Comment (seanshpark) [inline compiler/one-cmds/tests/one-build_004.test:36]: after this, `../bin/dummy-compile` will exist and if something goes wrong, `../bin/dummy-compile` won't be removed.
maybe add a command to silently remove this file in `trap_err_onexit` ?

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**